### PR TITLE
Android tasks

### PR DESCRIPTION
**Pull request by kunall17**
Currently I have converted the document into md format, 
For the script do I have to write each of 42 tasks in this format ? 
```
    upload_task(
        # https://developers.google.com/open-source/gci/resources/downloads/TaskAPISpec.pdf
        name = 'Cleanup Zulip code to follow PEP-8 style standards',
        description = description % rule,
        status = 1, # 1: draft, 2: published
        max_instances = 1,
        mentors = ['niftynei@gmail.com', 'rishig@zulipchat.com', 'tabbott@zulipchat.com'],
        tags = ['python', 'pep8'], # free text
        is_beginner = rule in beginner_rules,
        # 1: Coding, 2: User Interface, 3: Documentation & Training,
        # 4: Quality Assurance, 5: Outreach & Research
        categories = [1, 4],
        time_to_complete_in_days = 3, # must be between 3 and 7
        # Field currently not accessible via API. gci-support says it is coming soon.
        # external_url = "TODO",
        private_metadata = "PEP8",
        do_upload = args.force)
```

### CODE REVIEW

**Comment by rishig** [comment]
@kunall17 pep8.py and mypy-annotations.py are good files to look at -- it should only be one or two such blocks of code for the whole category.

**Comment by rishig** [comment]
ah, now that look more closely, it's likely we'll want to go with a different strategy here. Let me think and get back to you.

**Comment by kunall17** [comment]
Yeah, because the pep8 is for particular **type** of tasks and others as well, and this android comprises of documentation + code + translation in different tasks! 

Thanks for helping on this @rishig :)

**Comment by rishig** [comment]
Thanks for compiling these @kunall17! A few notes:
* The translation ones can just be combined with the general translation category, since the android translations should (eventually, if not now) just be sent to transifex.
* Out of the rest, I would recommend choosing 5 code tasks to put up for Monday. The task description can be something like 
"""
Write code for the Zulip Android project! 

Instructions for Android tasks are at
https://github.com/zulip/zulip-gci/blob/master/tasks/android.md.

Resolve the task titled **Use an alternate method for ViewPropertyAnimator.start()**.
"""
`user-guides.py` might be a good one to follow. 

You can then add more tasks later on Monday or later in the week as you get a feel for the experience of the students / your mentoring capacity. (Unlike some of the other task categories, I suspect most of the non-android mentors won't be able to check off on these tasks.) 
* Take a look at some of the other '.md' files to get a sense of the general format and content needed. E.g. prerequisites, where to submit the PR, etc. These have all been updated pretty recently, so it's possible they didn't have all this information when you last looked.

File: scripts/android.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+for title, descriptive_text, checklist in tasks:
+    upload_task(
+        # https://developers.google.com/open-source/gci/resources/downloads/TaskAPISpec.pdf
+        name = title,
+        description = description % (descriptive_text, checklist),
+        status = 1, # 1: draft, 2: published
+        max_instances = 1,
+        mentors = ['niftynei@gmail.com', 'rishig@zulipchat.com', 'tabbott@zulipchat.com', 'kunall.gupta17@gmail.com'],
```
Comment (kunall17) [inline scripts/android.py]: Not really sure whom all to include in the mentors therefore i just added my email there!

**Comment by kunall17** [comment]
@rishig  For the current week starting from monday I made a script only for the documentation part here! 
Directly jumping into bug fixing or coding new features could be really hectic at first! :)

**Comment by rishig** [comment]
Sounds good! Could you make a android.md file with just the tasks you are planning to post for Monday? I would remove the Task numbers, so that it is easy to add and remove tasks without having to change all the numbers.

File: scripts/android.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        tags = ['documentation', 'user guides'], # free text
+        is_beginner = True,
+        # 1: Coding, 2: User Interface, 3: Documentation & Training,
+        # 4: Quality Assurance, 5: Outreach & Research
+        categories = [3, 5],
+        time_to_complete_in_days = 3, # must be between 3 and 7
+        # Field currently not accessible via API. gci-support says it is coming soon.
+        external_url = "https://github.com/zulip/zulip-gci/blob/master/tasks/android.md",
+        private_metadata = "user-guides-A",
```
Comment (rishig) [inline scripts/android.py]: This should be "android-A".

File: scripts/android.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    ("API routes used in the application (for developers)", "the API routes by the Application.\nYou can get the information about these API’s at https://zulip.readthedocs.io/", """All the API’s used like get_auth_backends and information about the API’s."""),
+]
+
+description = """A descriptive text/Documentation and compiled like https://zulip.readthedocs.io/ describing """ + "%s**\n\n **Checklist**\n %s"
+
+
+for title, descriptive_text, checklist in tasks:
+    upload_task(
+        # https://developers.google.com/open-source/gci/resources/downloads/TaskAPISpec.pdf
+        name = title,
```
Comment (rishig) [inline scripts/android.py]: I would make all the titles "Write documentation for the Zulip android project"

File: scripts/android.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    ("some miscellaneous features", "some features.", """Day/Night mode\nFiltering messages\nMore """),
+    ("widget", " the usage with screenshots of the widget.", """Update time of receiving messages.\nNarrowed view in the widget."""),
+    ("menu of long click on message", " the options available when long press the message.\nWith a screenshot of the menu.", """Include documentation about each of the options."""),
+    ("supported backends", "the implemented backends in the app.\n You can get the information about these API’s at https://zulip.readthedocs.io/ ", """Include text about the 3 Authentication login backend’s the project handles currently which are Email, Google, DevAuth Backend’s."""),
+    ("API routes used in the application (for developers)", "the API routes by the Application.\nYou can get the information about these API’s at https://zulip.readthedocs.io/", """All the API’s used like get_auth_backends and information about the API’s."""),
+]
+
+description = """A descriptive text/Documentation and compiled like https://zulip.readthedocs.io/ describing """ + "%s**\n\n **Checklist**\n %s"
```
Comment (rishig) [inline scripts/android.py]: `user-guides.py` is a good one to copy here. So something like
```
description = """Good user guides help users and search engines discover Zulip features.
Help write user guides for the Zulip Android project!

Instructions for all Android tasks are at
https://github.com/zulip/zulip-gci/blob/master/tasks/android.md.

"""

description_A = description + "For this task, add a user guide for the following feature: **%s**"
```

File: scripts/android.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import argparse
+from upload import upload_task
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-f', '--force', dest='force', action="store_true", default=False)
+args = parser.parse_args()
+
+tasks = [
+    ("Android Project", "different parts of the Application in detail.", """Write about the basic functionality of the app, how the app should be used. Covering things narrowing to a stream or narrowing to a topic is a must include."""),
```
Comment (rishig) [inline scripts/android.py]: This is probably too broad.

File: scripts/android.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    ("Chat Box", "chatBox (functioning and what each button does) with screenshots.", """Include text about the behavior details about swipe to remove and it automatically goes away when the main recyclerView is scrolled.\nSwitching between private and stream mode. """),
+    ("Streams Drawer", "the streams drawer (with screenshots)", """Unread counts (Currently there is a bug which is breaking the unread counts, but I hope it would be fixed soon) and it expands and shows the topics of that stream. On clicking the Stream name you can directly narrow to that stream. """),
+    ("some miscellaneous features", "some features.", """Day/Night mode\nFiltering messages\nMore """),
```
Comment (rishig) [inline scripts/android.py]: I think it is fine to separate these out into separate tasks.

File: scripts/android.py (hunk 4 above)
Comment (rishig) [inline scripts/android.py]: (and then the detailed instructions, etc. go in the .md file. Check out user-guides.md for an example.)

File: scripts/android.py (hunk 6 above)
Comment (kunall17) [inline scripts/android.py]: Like different upload_task?

File: scripts/android.py (hunk 1 above)
Comment (rishig) [inline scripts/android.py]: yup, perfect. We haven't decided the list of mentors yet, so we'll just update this on Sunday.

File: scripts/android.py (hunk 5 above)
Comment (rishig) [inline scripts/android.py]: yeah, I think for initial GCI tasks the more narrowly scoped the better.

File: scripts/android.py (hunk 6 above)
Comment (rishig) [inline scripts/android.py]: yeah. (Each call to upload_task creates a new task at codein.withgoogle.com.)

File: scripts/android.py (hunk 3 above)
Comment (kunall17) [inline scripts/android.py]: Changed the titles according to the tasks! :)

File: scripts/android.py (hunk 5 above)
Comment (kunall17) [inline scripts/android.py]: Removed..

**Comment by kunall17** [comment]
Updated the branch!

**Comment by kunall17** [comment]
Done :)

File: scripts/android.py [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+tags = ['documentation', 'user guides'], # free text
+is_beginner = True,
+# 1: Coding, 2: User Interface, 3: Documentation & Training,
+# 4: Quality Assurance, 5: Outreach & Research
+categories = [3, 5],
+time_to_complete_in_days = 3, # must be between 3 and 7
+# Field currently not accessible via API. gci-support says it is coming soon.
+external_url = "https://github.com/zulip/zulip-gci/blob/master/tasks/android.md",
+private_metadata = "android-A",
+do_upload = args.force)
```
Comment (timabbott) [inline scripts/android.py]: this seems to be teh same as the previous one?

Also maybe we should add `android` to all the tags?

**Comment by kunall17** [comment]
Updated!

**Comment by timabbott** [comment]
@kunall17 I'm really sorry we never got around to reviewing and pushing this; things just got too crazy with the first weeks of GCI.  I think we can reuse these as starter projects for all the interested GSoC students, though; do you want to just move them to new GitHub issues for the Android project?  

Sorry again I dropped the ball on this!

**Comment by kunall17** [comment]
:(
Hopefully we can use this in future, i have more of them!

**Pull request closed without merging**